pydantic/generics.py

```diff
@@ -99,15 +107,48 @@ def __concrete_name__(cls: Type[Any], params: Tuple[Type[Any], ...]) -> str:
         """
         This method can be overridden to achieve a custom naming scheme for GenericModels
         """
-        param_names = [param.__name__ if hasattr(param, '__name__') else str(param) for param in params]
+        try:
+            GenericAlias = typing.GenericAlias  # type: ignore
+        except AttributeError:
+            GenericAlias = ()  # for all python < 3.9
+        param_names = [
+            param.__name__ if hasattr(param, '__name__') and not isinstance(param, GenericAlias) else str(param)
+            for param in params
+        ]
         params_component = ', '.join(param_names)
         return f'{cls.__name__}[{params_component}]'
 
 
-def resolve_type_hint(type_: Any, typevars_map: Dict[Any, Any]) -> Type[Any]:
-    if get_origin(type_) and getattr(type_, '__parameters__', None):
-        concrete_type_args = tuple([typevars_map[x] for x in type_.__parameters__])
-        return type_[concrete_type_args]
+def is_identity_typevars_map(typevars_map: Dict[TypeVarType, Type[Any]]) -> bool:
```

Comment:
docstring please.

Reply:
I moved this to a more generic "all_identical" function (moved that to utils) and added some docs/examples/tests.  Hope it's clearer now.